 cmake/FindIconv.cmake | 26 +++++++++++++-------------
 config.h              |  2 +-
 2 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/cmake/FindIconv.cmake b/cmake/FindIconv.cmake
index 24cdae3..a13ce40 100644
--- a/cmake/FindIconv.cmake
+++ b/cmake/FindIconv.cmake
@@ -23,20 +23,20 @@ ENDIF(ICONV_INCLUDE_DIR AND ICONV_LIBRARIES)
 
 set(CMAKE_REQUIRED_INCLUDES ${ICONV_INCLUDE_DIR})
 set(CMAKE_REQUIRED_LIBRARIES ${ICONV_LIBRARIES})
-IF(ICONV_FOUND)
-  check_cxx_source_compiles("
+#IF(ICONV_FOUND)
+#  check_cxx_source_compiles("
   #include <iconv.h>
-  int main(){
-    iconv_t conv = 0;
-    const char* in = 0;
-    size_t ilen = 0;
-    char* out = 0;
-    size_t olen = 0;
-    iconv(conv, &in, &ilen, &out, &olen);
-    return 0;
-  }
-" ICONV_SECOND_ARGUMENT_IS_CONST )
-ENDIF(ICONV_FOUND)
+#  int main(){
+#    iconv_t conv = 0;
+#    const char* in = 0;
+#    size_t ilen = 0;
+#    char* out = 0;
+#    size_t olen = 0;
+#    iconv(conv, &in, &ilen, &out, &olen);
+#    return 0;
+#  }
+#" ICONV_SECOND_ARGUMENT_IS_CONST )
+#ENDIF(ICONV_FOUND)
 set(CMAKE_REQUIRED_INCLUDES)
 set(CMAKE_REQUIRED_LIBRARIES)
 
diff --git a/config.h b/config.h
index 2a96a0d..e54156f 100644
--- a/config.h
+++ b/config.h
@@ -32,7 +32,7 @@
 #    define W_OK 0x02
 #endif
 
-#define snprintf _snprintf
+//#define snprintf _snprintf
 /* Font configuration directory */
 #define CONFDIR "c:/windows/etc"
 
