diff --git "a/CMakeLists.txt" "b/CMakeLists.txt"
index 83b39d6..20f43b5 100644
--- "a/CMakeLists.txt"
+++ "b/CMakeLists.txt"
@@ -12,17 +12,17 @@ endif(NOT CMAKE_BUILD_TYPE)
 
 include ( GNUInstallDirs )
 
-option(BUILD_SHARED_LIBS "Build a shared library" ON)
-option(BUILD_STATIC_LIBS "Build the static library" OFF)
+option(BUILD_SHARED_LIBS "Build a shared library" OFF)
+option(BUILD_STATIC_LIBS "Build the static library" ON)
 option(BUILD_TESTS "Build test programs" OFF)
 option(MSVC_STATIC_RUNTIME "Build with static runtime libs (/MT)" ON)
 option(ZZIPMMAPPED "Build libzzipmmapped (not fully portable)" ON)
 option(ZZIPFSEEKO "Build libzzipfseeko (based on posix.1 api)" ON)
 option(ZZIPWRAP "Build libzzipwrap (non-installed example)" ON)
 option(ZZIPSDL "Build zzcatsdl (includable example code)" ON)
-option(ZZIPBINS "Build bins/unzzip (unzip-like executables)" ON)
-option(ZZIPTEST "Build test/zzipself (self-extract example)" ON)
-option(ZZIPDOCS "Build docs/manpages (and website content)" ON)
+option(ZZIPBINS "Build bins/unzzip (unzip-like executables)" OFF)
+option(ZZIPTEST "Build test/zzipself (self-extract example)" OFF)
+option(ZZIPDOCS "Build docs/manpages (and website content)" OFF)
 
 add_subdirectory(zzip)
 
